include/seastar/rpc/rpc.hh

```diff
@@ -674,6 +706,10 @@ protected:
 ///
 /// ## Configuration
 ///
+/// The object returned by `make_client()` contains an `opts` member of type
+/// \ref seastar::rpc::send_options, which can be modified (by setting its fields)
+/// before executing RPCs using this object.
```

Comment:
The object suppose to be a transparent functor that is only used as such. The alternative is to change `make_client()` to receive `send_option` somehow.

Reply:
For me a "functor" is an object that emulates function behaviour. Lambda is a syntactic sugar to create one.

You have another patch that is simple and approved. I do not see why continue with this approach.